src/Workspaces/Core/Portable/Workspace/Workspace.cs

```diff
@@ -1557,6 +1563,17 @@ protected virtual void ApplyCompilationOptionsChanged(ProjectId projectId, Compi
             this.OnCompilationOptionsChanged(projectId, options);
         }
 
+        /// <summary>
+        /// This method is called during <see cref="TryApplyChanges(Solution)"/> to change the default namespace.
+        ///
+        /// Override this method to implement the capability of changing default namespace.
+        /// </summary>
+        protected virtual void ApplyDefaultNamespaceChanged(ProjectId projectId, string defaultNamespace)
```

Comment:
```suggestion
        protected virtual void ApplyDefaultNamespaceChanged(ProjectId projectId, string? defaultNamespace)
```

Since the default namespace can be null.